packages/swapper/src/swappers/osmosis/OsmosisSwapper.ts

```diff
@@ -0,0 +1,410 @@
+    const buyAsset = args.trade.buyAsset
+    let sellAmount = args.trade.sellAmount
+    const buyAmount = args.trade.buyAmount
+    const wallet = args.wallet
+
+    if (!sellAsset) throw Error('missing sellAsset')
+    if (!buyAsset) throw Error('missing buyAsset')
+    if (!sellAmount) throw Error('missing sellAmount')
+    if (!buyAmount) throw Error('missing buyAmount')
+    if (!wallet) throw Error('missing wallet')
+
+    const pair = sellAsset.symbol + '_' + buyAsset.symbol
+    const sellAssetDenom = symbolDenomMapping[sellAsset.symbol as keyof IsymbolDenomMapping]
+    const buyAssetDenom = symbolDenomMapping[buyAsset.symbol as keyof IsymbolDenomMapping]
+
+    const gas = '1350000'
+
+    const osmosisAdapter = this.deps.adapterManager.get(osmosisChainId) as
+      | osmosis.ChainAdapter
+      | undefined
+    const cosmosAdapter = this.deps.adapterManager.get(cosmosChainId) as
+      | cosmos.ChainAdapter
+      | undefined
+
+    if (cosmosAdapter && osmosisAdapter) {
+      const osmoAddress = await (wallet as OsmosisWallet).osmosisGetAddress({
+        addressNList: bip32ToAddressNList("m/44'/118'/0'/0/0")
+      })
+      const atomAddress = await (wallet as CosmosWallet).cosmosGetAddress({
+        addressNList: bip32ToAddressNList("m/44'/118'/0'/0/0")
+      })
+
+      if (!osmoAddress) throw Error('Failed to get osmoAddress!')
+      if (!atomAddress) throw Error('Failed to get atomAddress!')
+
+      let sellAddress
+      let buyAddress
+
+      if (pair === 'ATOM_OSMO') {
+        sellAddress = atomAddress
+        buyAddress = osmoAddress
+
+        const transfer = {
+          sender: sellAddress,
+          receiver: buyAddress,
+          amount: String(sellAmount)
+        }
+
+        const { tradeId } = await this.performIbcTransfer(
+          transfer,
+          cosmosAdapter,
+          wallet,
+          this.deps.cosmosUrl,
+          this.deps.osmoUrl,
+          'uatom',
+          'channel-141',
+          '0'
+        )
+
+        // wait till confirmed
+        const pollResult = await pollForComplete(tradeId, this.deps.cosmosUrl)
+        if (pollResult !== 'success') throw new Error('ibc transfer failed')
+
+        sellAmount = await pollForAtomChannelBalance(buyAddress, this.deps.osmoUrl)
+      } else if (pair === 'OSMO_ATOM') {
+        sellAddress = osmoAddress
+        buyAddress = atomAddress
+      } else {
+        throw Error('Pair not supported! ' + pair)
+      }
+
+      if (!sellAddress) throw new Error('no sell address')
+      if (!buyAddress) throw new Error('no buy address')
+
+      const sender = osmoAddress
+      const accountUrl = `${this.deps.osmoUrl}/auth/accounts/${sender}`
+      const responseAccount = await axios.get(accountUrl)
+      const accountNumber = responseAccount.data.result.value.account_number || 0
+      const sequence = responseAccount.data.result.value.sequence || 0
+      const osmoAddressNList = bip32ToAddressNList("m/44'/118'/0'/0/0")
```

Comment:
all these paths should involve the sell asset account number and / or buy asset account number. its pointless to pass it in and then just hardcode like this. see other swappers